File: dotnet/src/Downstairs.Infrastructure/Persistence/Models/Order.cs
using System;
using System.Collections.Generic;

namespace Downstairs.Infrastructure.Persistence.Models;

public partial class Order
{
    public long Id { get; set; }

    public long UserId { get; set; }

    public long CustomerId { get; set; }

    public long? ServiceId { get; set; }

    public long? SubscriptionId { get; set; }

    public long? InvoiceId { get; set; }

    public long? OrderFixedPriceId { get; set; }

    public string OrderableType { get; set; } = null!;

    public long OrderableId { get; set; }

    public string Status { get; set; } = null!;

    public string PaidBy { get; set; } = null!;

    public DateTime? PaidAt { get; set; }

    public DateTime OrderedAt { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public DateTime? DeletedAt { get; set; }

    public virtual Customer Customer { get; set; } = null!;

    public virtual Invoice? Invoice { get; set; }

    public virtual OrderFixedPrice? OrderFixedPrice { get; set; }

    public virtual ICollection<OrderRow> OrderRows { get; set; } = new List<OrderRow>();

    public virtual Service? Service { get; set; }

    public virtual Subscription? Subscription { get; set; }

    public virtual User User { get; set; } = null!;
}

File: php-laraval-app/app/Models/Order.php
<?php

namespace App\Models;

use App\Enums\Order\OrderStatusEnum;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\MorphTo;
use Illuminate\Database\Eloquent\SoftDeletes;

class Order extends Model
{
    use HasFactory;
    use SoftDeletes;

    /**
     * Indicates if the model should be timestamped.
     *
     * @var bool
     */
    public $timestamps = true;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'user_id',
        'customer_id',
        'service_id',
        'subscription_id',
        'invoice_id',
        'order_fixed_price_id',
        'status',
        'paid_by',
        'paid_at',
        'ordered_at',
        'orderable_type',
        'orderable_id',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array
     */
    protected $casts = [
        'paid_at' => 'datetime',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
        'deleted_at' => 'datetime',
        'ordered_at' => 'datetime',
    ];

    /**
     * Define the alias of the columns or relations.
     *
     * @var array<string,string>
     */
    protected array $aliases = [
        'schedule' => 'orderable',
    ];

    /**
     * Define the models for polymorphic relations.
     *
     * @var array<string,string[]>
     */
    protected array $relationsMorphMap = [
        'orderable' => [Schedule::class, LaundryOrder::class, StoreSale::class],
    ];

    /**
     * Order relation to user model.
     */
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class)->withTrashed();
    }

    /**
     * Order relation to customer model.
     */
    public function customer(): BelongsTo
    {
        return $this->belongsTo(Customer::class)->withTrashed();
    }

    /**
     * Order relation to service model.
     */
    public function service(): BelongsTo
    {
        return $this->belongsTo(Service::class)->withTrashed();
    }

    /**
     * Order relation to order row model.
     */
    public function rows(): HasMany
    {
        return $this->hasMany(OrderRow::class);
    }

    /**
     * Get the parent.
     */
    public function orderable(): MorphTo
    {
        return $this->morphTo('orderable')->withTrashed();
    }

    /**
     * Get the invoice.
     */
    public function invoice(): BelongsTo
    {
        return $this->belongsTo(Invoice::class)->withTrashed();
    }

    /**
     * Order relation to subscription model.
     */
    public function subscription(): BelongsTo
    {
        return $this->belongsTo(Subscription::class)->withTrashed();
    }

    public function fixedPrice(): BelongsTo
    {
        return $this->belongsTo(OrderFixedPrice::class, 'order_fixed_price_id')->withTrashed();
    }

    public function scopeDraft(Builder $query): Builder
    {
        return $query->where('status', OrderStatusEnum::Draft());
    }
}
